src/test/java/KeyProviderTest.java

```diff
@@ -0,0 +1,57 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import com.google.sps.utility.KeyProvider;
+import java.io.IOException;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Test Key Provider functions */
+@RunWith(JUnit4.class)
+public class KeyProviderTest {
```

Comment:
I'm not sure this class even needs tests, however if it does (and it probably does), it will need to follow a similar pattern to the client servlets. Unit tests should not involve IO operations as this is an external dependency. To test this, you would need to use dependency injection, where some client would do the read from the hard drive, and you would only test the logic of transforming the rawJson to extracting the actual keys.

Reply:
This is a great point! We talked about this in the standup today and I have changed the tests to only handle the logic of transforming json to the actual values of the keys. Thanks!